pkg/core/inject/inject.go

```diff
@@ -31,13 +32,34 @@ func injectDependency(
 
 	numField := t.NumField()
 	for i := 0; i < numField; i++ {
-		dependencyName := t.Field(i).Tag.Get("dependency")
-		if dependencyName == "" {
+		dependencyTag := t.Field(i).Tag.Get("dependency")
+		if dependencyTag == "" {
 			continue
 		}
 
+		dependencyArgs := strings.Split(dependencyTag, ",")
+		dependencyName := dependencyArgs[0]
+		options := dependencyArgs[1:]
+		optionSet := arrayToSet(options)
```

Comment:
Why optionSet?

I think the arrayToMap that return `map[string]bool` will solve the problem.

Reply:
https://www.davidkaya.sk/2017/12/10/sets-in-golang/

> However I prefer the first variant since struct{} takes 0 bytes

Reply:
💯